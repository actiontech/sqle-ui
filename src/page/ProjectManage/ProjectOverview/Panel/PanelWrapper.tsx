import { Card } from 'antd';
import { PanelWrapperProps } from './index';

const PanelWrapper: React.FC<PanelWrapperProps> = ({
  children,
  title,
  subTitle,
  loading,
  error,
  bodyStyle,
}) => {
  return (
    <Card
      title={title}
      extra={subTitle}
      hoverable={true}
      loading={loading}
      type="inner"
      bordered
      bodyStyle={bodyStyle}
      style={{
        height: '100%',
      }}
    >
      {error ? error : children}
    </Card>
  );
};

export default PanelWrapper;
