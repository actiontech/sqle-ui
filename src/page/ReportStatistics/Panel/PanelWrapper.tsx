import { Card } from 'antd';
import { PanelWrapperProps } from './index';

const PanelWrapper: React.FC<PanelWrapperProps> = ({
  children,
  title,
  subTitle,
  loading,
  error,
}) => {
  return (
    <Card
      title={title}
      extra={subTitle}
      hoverable={true}
      loading={loading}
      type="inner"
      bordered
      style={{
        height: '100%',
      }}
    >
      {error ? error : children}
    </Card>
  );
};

export default PanelWrapper;
