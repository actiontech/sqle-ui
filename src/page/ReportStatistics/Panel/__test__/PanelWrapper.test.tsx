import { render, screen } from '@testing-library/react';
import { PanelWrapperProps } from '..';
import PanelWrapper from '../PanelWrapper';

const defaultConfig: PanelWrapperProps = {
  title: 'Title',
  subTitle: 'subTitle',
  loading: false,
  error: null,
};

const children = <span>children</span>;

describe('test PanelWrapper', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <PanelWrapper {...defaultConfig}>{children}</PanelWrapper>
    );

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when loading is equal true', () => {
    const { container } = render(
      <PanelWrapper {...{ ...defaultConfig, loading: true }}>
        {children}
      </PanelWrapper>
    );

    expect(container).toMatchSnapshot();
  });

  test('should render error when error is not undefined', () => {
    const { container } = render(
      <PanelWrapper
        {...{ ...defaultConfig, error: <span>error message</span> }}
      >
        {children}
      </PanelWrapper>
    );

    expect(screen.queryByText('children')).not.toBeInTheDocument();
    expect(screen.getByText('error message')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
