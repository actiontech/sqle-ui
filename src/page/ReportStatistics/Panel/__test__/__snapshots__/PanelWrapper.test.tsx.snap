// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PanelWrapper should match snapshot 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner"
    style="height: 100%;"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Title
        </div>
        <div
          class="ant-card-extra"
        >
          subTitle
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <span>
        children
      </span>
    </div>
  </div>
</div>
`;

exports[`test PanelWrapper should match snapshot when loading is equal true 1`] = `
<div>
  <div
    class="ant-card ant-card-loading ant-card-bordered ant-card-hoverable ant-card-type-inner"
    style="height: 100%;"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Title
        </div>
        <div
          class="ant-card-extra"
        >
          subTitle
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-skeleton ant-skeleton-active"
      >
        <div
          class="ant-skeleton-content"
        >
          <ul
            class="ant-skeleton-paragraph"
          >
            <li />
            <li />
            <li />
            <li
              style="width: 61%;"
            />
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test PanelWrapper should render error when error is not undefined 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner"
    style="height: 100%;"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Title
        </div>
        <div
          class="ant-card-extra"
        >
          subTitle
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <span>
        error message
      </span>
    </div>
  </div>
</div>
`;
